File: scala_api/src/main/scala/application/dataAnalysis/operations/ReadFileOperation.scala
package application.dataAnalysis.operations

import application.dataAnalysis.Context.spark
import application.dataAnalysis.operations.Helpers.ReadWriteFiler.safeData
import application.dataAnalysis.operations.Helpers.StatisticsCreator
import application.models.statistics.DataStatistics

object ReadFileOperation extends StatisticsCreator {

  def apply(path: String): DataStatistics = {
    val file = {
      if (path.endsWith(".csv")) {
        spark.read
          .option("inferSchema", "true")
          .option("header", "true")
          .option("sep", ",")
          .csv(path)
          .na.drop()
      } else {
        spark.read
          .option("inferSchema", "true")
          .parquet(path)
      }
    }
    safeData(file)
    createStatisticsSummary(file)
  }

}

File: python_api/SparkVersion/Operations/ReadFileOperation.py
import Context
from SparkVersion.Operations.Helpers.ReadWriteFiler import safeData
from SparkVersion.Operations.Helpers.StatisticsCreator import createStatisticsSummary


def readFile(filename):
    file = Context.spark.read \
        .option("inferSchema", "true") \
        .option("header", "true") \
        .option("sep", ",") \
        .csv(Context.defaultDirectoryPath + filename) \
        .na.drop() if filename.endswith(".csv") else \
        Context.spark.read.option("inferSchema", "true").parquet(Context.defaultDirectoryPath + filename).na.drop()
    safeData(file)
    return createStatisticsSummary(file)
